src/install.rs

```diff
@@ -76,7 +76,12 @@ impl<'a> InstallMethod<'a> {
             (false, _) => UpdateStatus::Unchanged,
         };
 
-        Ok(status)
+        // Final check, to ensure we're installed
+        if !toolchain.exists() {
```

Comment:
When could this happen? I know for freebsd 1.8 it happened, but do you know why?

Reply:
Yes, it's when all the components are *present* in the channel, but they're marked unavailable for installation.

Reply:
Other approaches I tried were much more complex than this simple cross-check at the end.